library/waku_thread/waku_thread.nim

```diff
@@ -93,36 +73,31 @@ proc destroyWakuThread*(ctx: ptr WakuContext): Result[void, string] =
 
   joinThread(ctx.thread)
   ?ctx.reqSignal.close()
-  ?ctx.respSignal.close()
   freeShared(ctx)
 
   return ok()
 
 proc sendRequestToWakuThread*(
-    ctx: ptr WakuContext, reqType: RequestType, reqContent: pointer
-): Result[string, string] =
-  let req = InterThreadRequest.createShared(reqType, reqContent)
+    ctx: ptr WakuContext,
+    reqType: RequestType,
+    reqContent: pointer,
+    callback: WakuCallBack,
+    userData: pointer,
+): Result[void, string] =
+  let req = WakuThreadRequest.createShared(reqType, reqContent, callback, userData)
   ## Sending the request
   let sentOk = ctx.reqChannel.trySend(req)
   if not sentOk:
+    deallocShared(req)
     return err("Couldn't send a request to the waku thread: " & $req[])
 
   let fireSyncRes = ctx.reqSignal.fireSync()
   if fireSyncRes.isErr():
+    deallocShared(req)
     return err("failed fireSync: " & $fireSyncRes.error)
 
   if fireSyncRes.get() == false:
+    deallocShared(req)
     return err("Couldn't fireSync in time")
 
-  # Waiting for the response
-  let res = waitSync(ctx.respSignal)
-  if res.isErr():
-    return err("Couldnt receive response signal")
-
-  var response: ptr InterThreadResponse
-  var recvOk = ctx.respChannel.tryRecv(response)
-  if recvOk == false:
-    return err("Couldn't receive response from the waku thread: " & $req[])
-
-  ## Converting the thread-safe response into a managed/CG'ed `Result`
-  return InterThreadResponse.process(response)
+  ok()
```

Comment:
where does the dealloc happen in the ok case? why not always dealloc here in this thread?